src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -283,21 +388,105 @@ private int recordExitStatus(FilePath workspace, int code) throws IOException, I
 
     }
 
+    private static final class AgentInfo implements Serializable {
+        private final OsType os;
+        private final ArchType arch;
+        private final FilePath launcherPath;
+        private boolean launcherCached;
+
+        public AgentInfo(OsType os, ArchType arch, FilePath launcherPath) {
+            this.os = os;
+            this.arch = arch;
+            this.launcherPath = launcherPath;
+            this.launcherCached = false;
+        }
+
+        public OsType getOs() {
+            return os;
+        }
+
+        public ArchType getArch() {
+            return arch;
+        }
+
+        public FilePath getLauncherPath() {
+            return launcherPath;
+        }
+
+        public void setLauncherAvailability(boolean isCached) {
+            launcherCached = isCached;
+        }
+
+        public boolean isLauncherCached() {
+            return launcherCached;
+        }
+    }
+
+    private static final class GetAgentInfo extends MasterToSlaveCallable<AgentInfo,RuntimeException> {
+        private final FilePath nodeRoot;
+        private String launcherVersion;
+
+        GetAgentInfo(FilePath nodeRoot, String pluginVersion) {
+            this.nodeRoot = nodeRoot;
+            this.launcherVersion = pluginVersion;
+        }
+
+        @Override public AgentInfo call() throws RuntimeException {
+            OsType os;
+            if (Platform.isDarwin()) {
+                os = OsType.DARWIN;
+            } else if (Platform.current() == Platform.WINDOWS) {
+                os = OsType.WINDOWS;
+            } else if(Platform.current() == Platform.UNIX && System.getProperty("os.name").equals("z/OS")) {
+                os = OsType.ZOS;
+            } else {
+                os = OsType.UNIX; // Default Value
+            }
+
+            // Note: This will only determine the architecture of the JVM. The result will be "32" or "64".
+            ArchType arch;
+            String bits = System.getProperty("sun.arch.data.model");
+            if (bits.equals("64")) {
+                arch = ArchType._64;
+            } else {
+                arch = ArchType._32; // Default Value
+            }
+
+            String launcherBinary = LAUNCHER_PREFIX + launcherVersion + "_" + os.getNameForBinary() + arch;
+            FilePath launcherPath = nodeRoot.child(launcherBinary);
+            AgentInfo agentInfo = new AgentInfo(os, arch, launcherPath);
+            try {
+                agentInfo.setLauncherAvailability(launcherPath.exists());
+            } catch (IOException | InterruptedException e) {
+                e.printStackTrace();
```

Comment:
I think you should throw the exception so that the master sees the failure. If you do that, it probably makes sense to change the signature of the class to `MasterToSlaveCallable<AgentInfo,Exception>` so that you don't have to wrap the exception.